### ISSUE TITLE
Support for Topics

### ISSUE THREAD

**Issue by jakejscott** [momentohq/client-sdk-dotnet#437]
Hey 👋 just wondering if the dotnet sdk has pub/sub support planned? 

Also will it support async enumerable on the subscriptions?

**Comment by malandis** [comment]
Hello Jake! 👋 

Pub/sub is definitely on the .NET SDK roadmap.

> Also will it support async enumerable on the subscriptions?

We haven't begun the design yet, but we will take into account when we do. Is there a reference pattern or implementation you have in mind?

**Comment by jakejscott** [comment]
Hi @malandis 

In my case I'd like to be able to use server sent events where I subscribe to a topic and as messages arrive I stream them in the response. Something along the lines of this:

```cs
using System.Runtime.CompilerServices;
using System.Text.Json;

public static class Program
{
    public static void Main(string[] args)
    {
        var builder = WebApplication.CreateBuilder(args);
        builder.Logging.AddSimpleConsole(options => { options.SingleLine = true; });       

        var app = builder.Build();

        app.MapGet("/", () => "Hello World!");

        app.MapGet("/sse", async (HttpContext context) =>
        {
            context.Response.StatusCode = 200;
            context.Response.ContentType = "text/event-stream";
            context.Response.Headers.CacheControl = "no-cache";
            context.Response.Headers.Connection = "keep-alive";
            
            var ct = context.RequestAborted;

            try
            {
                long id = 0;
                var topicName = "my-event";

                var subscription = SubscribeTopic(topicName, ct);
                
                await foreach (var item in subscription.WithCancellation(ct))
                {
                    id++;
                    await context.Response.WriteAsync(text: $"id:{id}\nevent: {topicName}\ndata: {JsonSerializer.Serialize(item)}\n\n", cancellationToken: ct);
                }
            }
            catch (OperationCanceledException)
            {
            }
        });

        app.Run();
    }

    private static async IAsyncEnumerable<object> SubscribeTopic(string topicName, [EnumeratorCancellation] CancellationToken cancellation)
    {
        while (!cancellation.IsCancellationRequested)
        {
            await Task.Delay(1000, cancellation);
            
            yield return new { guid = Guid.NewGuid() };
        }
    }
}
```

**Comment by cprice404** [comment]
Got another request for this via @allenheltondev today.

**Comment by cprice404** [comment]
Hello @jakejscott - FYI, we released v1.19.0 of our .NET SDK yesterday, which includes initial support for Topics. You can see the basic syntax here: 

https://github.com/momentohq/client-sdk-dotnet/blob/80d95f453c2c73f4057c5296d4fc22766cf39587/examples/TopicExample/Program.cs#L33

We are still finishing some remaining work on it, to make it do a better job of handling idle connections / disconnects from the server, etc., but the basic functionality is ready for consumption.

Would love to hear any feedback if you try it out!
